File: sdk_src/Server/MobileIMSDKServer_Open/src/net/x52im/mobileimsdk/server/protocal/c/PLoginInfo.java
/*
 * Copyright (C) 2021  即时通讯网(52im.net) & Jack Jiang.
 * The MobileIMSDK v6.1 Project. 
 * All rights reserved.
 * 
 * > Github地址：https://github.com/JackJiang2011/MobileIMSDK
 * > 文档地址：  http://www.52im.net/forum-89-1.html
 * > 技术社区：  http://www.52im.net/
 * > 技术交流群：320837163 (http://www.52im.net/topic-qqgroup.html)
 * > 作者公众号：“【即时通讯技术圈】”，欢迎关注！
 * > 联系作者：  http://www.52im.net/thread-2792-1-1.html
 *  
 * "即时通讯网(52im.net) - 即时通讯开发者社区!" 推荐开源工程。
 * 
 * PLoginInfo.java at 2021-8-4 21:24:15, code by Jack Jiang.
 */
package net.x52im.mobileimsdk.server.protocal.c;

public class PLoginInfo
{
	protected String loginUserId = null;
	protected String loginToken = null;
	protected String extra = null;
	protected long firstLoginTime = 0;
	
	public PLoginInfo(String loginUserId, String loginToken)
	{
		this(loginUserId, loginToken, null);
	}
	
	public PLoginInfo(String loginUserId, String loginToken, String extra)
	{
		this.loginUserId = loginUserId;
		this.loginToken = loginToken;
		this.extra = extra;
	}
	
	public String getLoginUserId()
	{
		return loginUserId;
	}

	public void setLoginUserId(String loginUserId)
	{
		this.loginUserId = loginUserId;
	}

	public String getLoginToken()
	{
		return loginToken;
	}

	public void setLoginToken(String loginToken)
	{
		this.loginToken = loginToken;
	}

	public String getExtra()
	{
		return extra;
	}
	
	public void setExtra(String extra)
	{
		this.extra = extra;
	}

	public long getFirstLoginTime()
	{
		return firstLoginTime;
	}

	public void setFirstLoginTime(long firstLoginTime)
	{
		this.firstLoginTime = firstLoginTime;
	}
	
	public static boolean isFirstLogin(long firstLoginTime)
	{
		return firstLoginTime <= 0;
	}
}

File: sdk_src/TCP_Client/MobileIMSDK4f_tcp_Open/lib/server/protocol/c/p_login_info.dart
import 'dart:core';

class PLoginInfo {
  String? _loginUserId;
  String? _loginToken;
  String? _extra;
  int _firstLoginTime = 0;

  PLoginInfo(String this._loginUserId, {String? loginToken, String? extra}) {
    _loginToken = loginToken;
    _extra = extra;
  }

  //命名式构造方法，也可以是用工厂构造方法
  PLoginInfo.fromJson(Map<String, dynamic> json)
      : _loginUserId = json['loginUserId'],
        _loginToken = json['loginToken'],
        _extra = json['extra'],
        _firstLoginTime = json['firstLoginTime'];

  //如果想写成协议，归档(json和对象互转时，为了使用方便)时，可以继承协议，那么可以用普通方法，而不是构造方法
  Map<String, dynamic> toJson() => {
        'loginUserId': _loginUserId,
        'loginToken': _loginToken,
        'extra': _extra,
        'firstLoginTime': _firstLoginTime
      };

  getLoginUserId() {
    return _loginUserId;
  }

  void setLoginUserId(String loginUserId) {
    _loginUserId = loginUserId;
  }

  getLoginToken() {
    return _loginToken;
  }

  void setLoginToken(String loginToken) {
    _loginToken = loginToken;
  }

  getExtra() {
    return _extra;
  }

  void setExtra(String extra) {
    _extra = extra;
  }

  getFirstLoginTime() {
    return _firstLoginTime;
  }

  void setFirstLoginTime(int firstLoginTime) {
    _firstLoginTime = firstLoginTime;
  }

  bool isFirstLogin() {
    return _firstLoginTime <= 0;
  }
}
